<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"> 
<html lang="de"> 
  <head> 
    <meta http-equiv="content-type" content="text/html; charset=utf-8"/> 
    <title>Open Object Store</title> 
  	<link rel="stylesheet" type="text/css" href="style.css" />
  </head> 
  <body>
    <div id="header">
      <a href="https://github.com/zussel/oos"><img style="z-order: 10000; float: left; top: 0; left: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_left_green_007200.png" alt="Fork me on GitHub"></a>
      <div id="navigation">
        <h1>OOS</h1>
        <p>Store all your objects in one container</p>
        <ul>
          <li><a href="../index.html">Home</a></li>
          <li class="selected">Documention</li>
          <li><a href="../download.html">Download</a></li>
          <!--li><a href="../contact.html">Contact</a></li-->
          <li><a href="../about.html">About</a></li>
        </ul>
      </div>
    </div>
    <div id="wrapper">
      <div id="content">
        <div>
<!-- Generated by Doxygen 1.8.6 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Overview</span></a></li>
      <li class="current"><a href="annotated.html"><span>Documentation</span></a></li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
      <li><a href="files.html"><span>File&#160;List</span></a></li>
      <li><a href="namespaces.html"><span>Namespace&#160;List</span></a></li>
    </ul>
  </div>
<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="namespaceoos.html">oos</a></li><li class="navelem"><a class="el" href="classoos_1_1unit__exception.html">unit_exception</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="classoos_1_1unit__exception-members.html">List of all members</a> &#124;
<a href="#pub-methods">Public Member Functions</a>  </div>
  <div class="headertitle">
<div class="title">oos::unit_exception Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Unit exception class thrown on a unit test assert definition.  
 <a href="classoos_1_1unit__exception.html#details">More...</a></p>

<p><code>#include &lt;unit_exception.hpp&gt;</code></p>

<p>Inherits std::exception.</p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a6e9410a09099c8c6c9ad9b279159e281"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classoos_1_1unit__exception.html#a6e9410a09099c8c6c9ad9b279159e281">unit_exception</a> (const std::string &amp;msg)  throw ()</td></tr>
<tr class="memdesc:a6e9410a09099c8c6c9ad9b279159e281"><td class="mdescLeft">&#160;</td><td class="mdescRight">Contructs a <a class="el" href="classoos_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a>.  <a href="#a6e9410a09099c8c6c9ad9b279159e281">More...</a><br/></td></tr>
<tr class="separator:a6e9410a09099c8c6c9ad9b279159e281"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1f6ea6136b661e24ae73af7cb3ee794d"><td class="memItemLeft" align="right" valign="top">virtual const char *&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classoos_1_1unit__exception.html#a1f6ea6136b661e24ae73af7cb3ee794d">what</a> () const   throw ()</td></tr>
<tr class="separator:a1f6ea6136b661e24ae73af7cb3ee794d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Unit exception class thrown on a unit test assert definition. </p>
<p>This <a class="el" href="classoos_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a> is thrown when an assert evaluates to false. This exception is caugt by the main <a class="el" href="classoos_1_1test__suite.html" title="The container for all unit tests. ">test_suite</a> object. </p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a6e9410a09099c8c6c9ad9b279159e281"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">oos::unit_exception::unit_exception </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&#160;</td>
          <td class="paramname"><em>msg</em></td><td>)</td>
          <td></td>
        </tr>
        <tr>
          <td align="right">throw </td><td>(</td><td colspan="2"></td>
        </tr>
        <tr>
          <td align="right"></td><td>)</td><td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Contructs a <a class="el" href="classoos_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a>. </p>
<p>A object of type <a class="el" href="classoos_1_1unit__exception.html" title="Unit exception class thrown on a unit test assert definition. ">unit_exception</a> with the given message is created.</p>
<dl class="params"><dt>Parameters</dt><dd>
  <table class="params">
    <tr><td class="paramname">msg</td><td>Message of the exception. </td></tr>
  </table>
  </dd>
</dl>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a1f6ea6136b661e24ae73af7cb3ee794d"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">virtual const char* oos::unit_exception::what </td>
          <td>(</td>
          <td class="paramname"></td><td>)</td>
          <td> const</td>
        </tr>
        <tr>
          <td align="right">throw </td><td>(</td><td colspan="2"></td>
        </tr>
        <tr>
          <td align="right"></td><td>)</td><td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">virtual</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<p>Returns the message of this exception.</p>
<dl class="section return"><dt>Returns</dt><dd>The message of the exception. </dd></dl>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>unit/unit_exception.hpp</li>
</ul>
</div><!-- contents -->
    <div class="clear" id="footer"></div>
  </div>
</body>
</html>
